### PR TITLE
Discourage use of hardcoded credentials

### PR DESCRIPTION
**Pull request by jdisanti**
## Motivation and Context
This fixes https://github.com/awslabs/smithy-rs/issues/317

## Checklist
- [ ] I have updated `CHANGELOG.md`
- [x] I have updated `aws/SDK_CHANGELOG.md` if applicable

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws/SDK_CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ vNext (Month Day, Year)
 
 **Breaking Changes**
 
+- `Credentials::from_keys` is now behind a feature flag named `hardcoded-credentials` in `aws-types`.
+  It is __NOT__ secure to hardcode credentials into your application, and the credentials
+  providers that come with the AWS SDK should be preferred. (smithy-rs#875, smithy-rs#317)
```
Comment (Velfi) [inline aws/SDK_CHANGELOG.md:11]: This is such a cool idea

**Review by Velfi — APPROVED** [review]

File: aws/rust-runtime/aws-types/src/credentials/credentials_impl.rs [hunk 2]
```diff
@@ -71,6 +76,36 @@ impl Credentials {
         }))
     }
 
+    /// Creates `Credentials` from hardcoded access key, secret key, and session token.
+    ///
+    /// _Note: In general, you should prefer to use the credential providers that come
+    /// with the AWS SDK to get credentials. It is __NOT__ secure to hardcode credentials
+    /// into your application. Only use this function if you really know what you're doing._
```
Comment (rcoh) [inline aws/rust-runtime/aws-types/src/credentials/credentials_impl.rs:83]: not sure about "if you know what you're doing" is the right sentiment. We should also probably suggest that people use `Credentials::new` if they're writing their own credential provider

File: aws/rust-runtime/aws-types/src/credentials/provider.rs [hunk 3]
```diff
@@ -138,22 +143,29 @@ impl Error for CredentialsError {
...
 pub mod future {
     use aws_smithy_async::future::now_or_later::NowOrLater;
     use std::future::Future;
     use std::pin::Pin;
     use std::task::{Context, Poll};
 
     type BoxFuture<'a, T> = Pin<Box<dyn Future<Output = T> + Send + 'a>>;
+
+    /// Convenience `ProvideCredentials` struct that implements the `ProvideCredentials` trait.
+    #[derive(Debug)]
```
Comment (rcoh) [inline aws/rust-runtime/aws-types/src/credentials/provider.rs:159]: this isn't right—this is the future newtype that `ProvideCredentials` must return

**Review by rcoh — APPROVED** [review]
LGTM! only major comment is on "if you know what you're doing." I wonder if there's a more concrete thing we could say there.

**Review by rcoh — APPROVED** [review]

**Pull request merged**